src/course-unit/sidebar/utils.jsx

```diff
@@ -0,0 +1,100 @@
+import {
+  CheckCircle as CheckCircleIcon,
+  CheckCircleOutline as CheckCircleOutlineIcon,
+  InfoOutline as InfoOutlineIcon,
+} from '@edx/paragon/icons';
+
+import { COLORS, UNIT_VISIBILITY_STATES } from '../constants';
+import messages from './messages';
+
+/**
+ * Get information about the publishing status.
+ * @param {Object} intl - The internationalization object.
+ * @param {boolean} hasChanges - Indicates if there are unpublished changes.
+ * @param {string} editedBy - The user who edited the content.
+ * @param {string} editedOn - The timestamp when the content was edited.
+ * @param {string} publishedBy - The user who last published the content.
+ * @param {string} publishedOn - The timestamp when the content was last published.
+ * @returns {string} Publish information based on the provided parameters.
+ */
+export const getPublishInfo = (intl, hasChanges, editedBy, editedOn, publishedBy, publishedOn) => {
+  if (hasChanges && editedOn && editedBy) {
+    return intl.formatMessage(messages.publishInfoDraftSaved, { editedOn, editedBy });
+  } if (publishedOn && publishedBy) {
+    return intl.formatMessage(messages.publishLastPublished, { publishedOn, publishedBy });
+  }
+
+  return intl.formatMessage(messages.publishInfoPreviouslyPublished);
```

Comment:
```suggestion
if (hasChanges && editedOn && editedBy) {
    return intl.formatMessage(messages.publishInfoDraftSaved, { editedOn, editedBy });
  } else if (publishedOn && publishedBy) {
    return intl.formatMessage(messages.publishLastPublished, { publishedOn, publishedBy });
  } else {
    return intl.formatMessage(messages.publishInfoPreviouslyPublished);
  }
```

Reply:
We have a linter configuration that will not allow us to do as you suggest